#1: Image Component Enhanced

BeeMargarida opened the pull request:
| - | - |
| --- | --- |
| Issue | Closes #1 |
| Dependencies | https://github.com/ripe-tech/ripe-sdk/pull/192 |
| Decisions | Image component with new props (crop, format) and initials logic (initialsBuilder, initialsGroup, showInitials and state). The state contains information about the initials and engraving. |
| Animated GIF | ![image_r_enh1](https://user-images.githubusercontent.com/25725586/89048662-0a2c4180-d348-11ea-9433-a995cf01dab1.gif)![image_r_enh](https://user-images.githubusercontent.com/25725586/89048640-04cef700-d348-11ea-8da7-ad842eaa101c.gif)|

Commit:
New props and support for showing personalization

gcandal commented:
Same thing as:
- https://github.com/ripe-tech/ripe-sdk-components-vue/pull/16#issuecomment-668688162
- https://github.com/ripe-tech/ripe-sdk-components-vue/pull/16#issuecomment-668688619

BeeMargarida commented:
> Same thing as:
> 
>     * [ripe-tech/ripe-sdk-components-vue#16 (comment)](https://github.com/ripe-tech/ripe-sdk-components-vue/pull/16#issuecomment-668688162)
> 
>     * [ripe-tech/ripe-sdk-components-vue#16 (comment)](https://github.com/ripe-tech/ripe-sdk-components-vue/pull/16#issuecomment-668688619)

@gcandal Regarding the testing, as I said in the vue PR, "its a bit more complicated than that. The setup of Ripe SDK is async, however it happens when the component is mounted, so there is no way in the test for waiting for the configuration to end. Its also very complicated by mocking the SDK because of the bindImage method, which must return an Image that implements Observable methods (bind, unbind, etc), which would involve an extensive mock".

Regarding the `crop` prop, it is working in react (used jpg to be more noticeable) with the this [fix](https://github.com/ripe-tech/ripe-sdk/pull/193).
![image_crop](https://user-images.githubusercontent.com/25725586/89387390-55ed3b00-d6fa-11ea-9223-7e43fff798fc.gif)

Commit:
Refactor and removing unnecessary logic

Removed partsData and using only props, responsability moved to the user.
Using promise in Ripe configuration, waiting for the configuration to end before binding the image.

BeeMargarida commented:
@gcandal Regarding the tests, even by using stubs and mocks, it is not possible. When mounting the component during the tests, the mounting returns before the async methods finish. It makes it impossible to tests behaviour related to the image, which only becomes available after the ripe sdk configuration ends.
